stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/OfflinePlugin.kt

```diff
@@ -60,6 +65,63 @@ public class OfflinePlugin(private val config: Config) : Plugin {
         logic = LogicRegistry(state)
     }
 
+    /**
+     * Method called when an edition in a message starts happens
+     *
+     * @param message [Message]
+     */
+    override suspend fun onMessageEditRequest(message: Message) {
+        val (channelType, channelId) = message.cid.cidToTypeAndId()
+        val channelLogic = logic.channel(channelType, channelId)
+
+        val isOnline = ChatDomain.instance().isOnline()
+
+        val messagesToEdit = message.copy(
+            updatedLocallyAt = Date(),
+            syncStatus = if (isOnline) SyncStatus.IN_PROGRESS else SyncStatus.SYNC_NEEDED
+        ).let(::listOf)
+
+        channelLogic.upsertMessages(messagesToEdit)
+        channelLogic.storeMessageLocally(messagesToEdit)
+    }
+
+    /**
+     * Method called when an edition in a message returns from the API
```

Comment:
.

Reply:
I'm going to create a script for that o.O